addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/ExtensionSelenium.java

```diff
@@ -634,9 +660,46 @@ private WebDriver getWebDriverImpl(
         if (proxyAddress == null) {
             return webDriverProviders.get(providedBrowser.getProviderId()).getWebDriver(requester);
         }
-        return webDriverProviders
-                .get(providedBrowser.getProviderId())
-                .getWebDriver(requester, proxyAddress, proxyPort);
+        WebDriver wd =
+                webDriverProviders
+                        .get(providedBrowser.getProviderId())
+                        .getWebDriver(requester, proxyAddress, proxyPort);
+        ExtensionScript extScript =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+        if (extScript != null) {
+            List<ScriptWrapper> scripts = extScript.getScripts(SCRIPT_TYPE_SELENIUM);
+            for (ScriptWrapper script : scripts) {
+                try {
+                    if (script.isEnabled()) {
+                        SeleniumScript s = extScript.getInterface(script, SeleniumScript.class);
+
+                        if (s != null) {
+                            new Thread("ZAP-selenium-script") {
+                                @Override
+                                public void run() {
+                                    s.browserLaunched(
```

Comment:
We should also handle/catch the Exception here.

Reply:
Doesnt the outer try-catch statement do this?

Reply:
No, this is already in the thread.

Reply:
Good point. Now done.